src/__tests__/index.test.ts

```diff
@@ -1,5 +1,420 @@
-describe('example test', () => {
-  it('works', () => {
-    expect(1 + 1).toEqual(2)
+import { ApiClient } from '../index'
+
+const fetchMock = jest.fn()
+global.fetch = fetchMock
+
+const mockResolvedOnce = (value: any) => {
+  fetchMock.mockReturnValueOnce(
+    Promise.resolve({
+      ok: true,
+      json: () => {
+        return Promise.resolve(value)
+      },
+      headers: {
+        get: () => {
+          return 200
+        },
+      },
+    }),
+  )
+}
+
+const mockFetchFail = () => {
+  fetchMock.mockReturnValueOnce(Promise.reject(new Error()))
+}
+
+const mockApiFail = () => {
+  fetchMock.mockReturnValueOnce(
+    Promise.resolve({
+      ok: false,
+      statusText: 'Wrong data format',
+      status: 400,
+    }),
+  )
+}
```

Comment:
Those look like useful utilites. Would it make sense to share them?

Reply:
Those utilities don't make sense to share because the consuming project needs to mock the apiclient and not the fetch call. However, it might make sense to export jest utils for the package itself so the consumer does not need to worry about it. Let me explore a bit :)

Reply:
Oh, I meant within the project. Since we would split the test files eventually